### PR TITLE
feat: pre-commit setup

### PR DESCRIPTION
**Pull request by micache**
Add pre-commit config and style-check file for the Github actions CI.

Running both on push or pull requests to any branches.

This config use black, isort, flake8 and autoflake to lint the code.

### CODE REVIEW

File: .github/workflows/style_check.yml
Comment (khankaholic) [inline .github/workflows/style_check.yml:1]: @micache We might only need to run this check whenever on:
- pushing to branch default `main` (not `master` in our repo) and `dev`
- creating PRs

Please help to re-config

Comment (khankaholic) [inline .github/workflows/style_check.yml:1]: @micache Created pre-commit by yourself, but then the check's failed. Did you forget any installation on local, lol?

**Review by khankaholic — CHANGES_REQUESTED** [review]

**Comment by khankaholic** [comment]
@micache Below is the PR template file, it's just simple to put it into folder `.github/` next to `workflows/` and create a commit for that. Afterwards, follow the template to edit your PR description.
[pull_request_template.md](https://github.com/user-attachments/files/16022853/pull_request_template.md)

**Comment by khankaholic** [comment]
@micache 
Another CI for PR lint validation, feel free to add this if you have time
Add this yaml file into `.github/workflows` and its config to the repo root (same level with `.pre-commit-config.yaml`)

yaml file
```txt
name: Lint PR

on:
  pull_request:
    types:
      - opened
      - edited
      - synchronize

permissions:
  pull-requests: write

jobs:
  pr-commitlint:
    name: Validate messages
    runs-on: ubuntu-latest
    permissions: write-all
    steps:
      - uses: wagoid/commitlint-github-action@v6
        id: lint_pr_title
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          configFile: ./.commitlintrc

      - uses: marocchino/sticky-pull-request-comment@v2
        # When the previous steps fails, the workflow would stop. By adding this
        # condition you can continue the execution with the populated error message.
        if: always() && (steps.lint_pr_title.outputs.results != null)
        with:
          header: pr-title-lint-error
          message: |
            Hey there and thank you for opening this pull request! 👋🏼
            We require pull request titles to follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/) and it looks like your proposed title needs to be adjusted.
            Details:
            ```
            ${{ steps.lint_pr_title.outputs.results }}
            ```
      # Delete a previous comment when the issue has been resolved
      - if: ${{ steps.lint_pr_title.outputs.error_message == null }}
        uses: marocchino/sticky-pull-request-comment@v2
        with:
          header: pr-title-lint-error
          delete: true
```

pr lint config
```
{
    "extends": ["@commitlint/config-conventional"],
    "rules": {
        "body-leading-blank": [1, "always"],
        "body-max-line-length": [2, "always", 100],
        "footer-leading-blank": [1, "always"],
        "footer-max-line-length": [2, "always", 100],
        "header-max-length": [2, "always", 100],
        "subject-case": [
            2,
            "never",
            []
        ],
        "subject-empty": [2, "never"],
        "subject-full-stop": [2, "never", "."],
        "type-case": [2, "always", "lower-case"],
        "type-empty": [2, "never"],
        "type-enum": [
            2,
            "always",
            [
                "build",
                "chore",
                "ci",
                "docs",
                "feat",
                "fix",
                "perf",
                "refactor",
                "revert",
                "style",
                "test"
            ]
        ]
    }
}
```

File: pr-lint-config.yaml
Comment (khankaholic) [inline pr-lint-config.yaml:1]: @micache rename this file as it is in the yml file
https://github.com/Croissant-Team-Cinnamon-Bootcamp-2024/homework-week-1/blob/44e2e18c9b467f4205295f589de6cd77ad6409f2/.github/workflows/lint_pr.yml#L24

**Review by khankaholic — CHANGES_REQUESTED** [review]

**Review by khankaholic — CHANGES_REQUESTED** [review]
@micache this action is a bit hard to use for our team at first.
You can consider to drop it.

File: pr-lint-config.yaml
Comment (khankaholic) [inline pr-lint-config.yaml:1]: @micache please remove this config file as well

**Review by khankaholic — CHANGES_REQUESTED** [review]

File: pr-lint-config.yaml
Comment (micache) [reply pr-lint-config.yaml:1]: done

**Review by khankaholic — APPROVED** [review]
LGTM

**Review by Huy1711 — APPROVED** [review]
LGTM

**Pull request merged**